### PR TITLE
Total count of users is displayed in direct messages.

### PR DESCRIPTION
**Pull request by lisakycho**
#### Summary
The number of members is displayed in the direct messages more modal.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/6521

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Has server changes https://github.com/mattermost/mattermost-server/pull/8847
- [x] Has redux changes https://github.com/mattermost/mattermost-redux/pull/508
- [x] Has UI changes

image

### CODE REVIEW

**Comment by hmhealey** [comment]
@lisakycho You've got some style issues that Jenkins is reporting
```
> eslint --ignore-pattern node_modules --ignore-pattern non_npm_dependencies --ignore-pattern dist --ext .js --ext .jsx . --quiet





/home/ubuntu/jenkins/workspace/mattermost-webapp_PR-1248-XPSPE67F5W4QRKV5WZVS2I55EOQVUU3SG2W2EWKCKVQNXR4DASTQ/components/more_direct_channels/index.js

  59:44  error  Missing trailing comma  comma-dangle



/home/ubuntu/jenkins/workspace/mattermost-webapp_PR-1248-XPSPE67F5W4QRKV5WZVS2I55EOQVUU3SG2W2EWKCKVQNXR4DASTQ/components/more_direct_channels/more_direct_channels.jsx

   64:15  error  'totalCount' is assigned a value but never used  no-unused-vars

  379:11  error  Trailing spaces not allowed                      no-trailing-spaces



/home/ubuntu/jenkins/workspace/mattermost-webapp_PR-1248-XPSPE67F5W4QRKV5WZVS2I55EOQVUU3SG2W2EWKCKVQNXR4DASTQ/components/multiselect/multiselect_list.jsx

  13:62  error  Block must not be padded by blank lines  padded-blocks



/home/ubuntu/jenkins/workspace/mattermost-webapp_PR-1248-XPSPE67F5W4QRKV5WZVS2I55EOQVUU3SG2W2EWKCKVQNXR4DASTQ/components/multiselect/multiselect.jsx

    9:20  error  'StatTypes' is defined but never used                   no-unused-vars

   17:58  error  Block must not be padded by blank lines                 padded-blocks

   21:37  error  Missing trailing comma                                  comma-dangle

   26:1   error  Expected indentation of 4 spaces but found 8            indent

  164:9   error  'memberCount' is never reassigned. Use 'const' instead  prefer-const

  167:31  error  There should be no space after '='                      react/jsx-equals-spacing

  170:54  error  Missing trailing comma                                  comma-dangle



✖ 11 problems (11 errors, 0 warnings)

  8 errors, 0 warnings potentially fixable with the `--fix` option.
```

**Comment by lisakycho** [comment]
@hmhealey, Sorry I ran out of time yesterday before I could fix it, but it should be good now! :)

File: components/multiselect/multiselect.jsx [hunk 1]
```diff
@@ -150,6 +155,17 @@ export default class MultiSelect extends React.Component {
             );
         }
 
+        const memberCount = (
+            <FormattedMessage
+                id='multiselect.numRemaining'
```
Comment (hmhealey) [inline components/multiselect/multiselect.jsx]: This needs its own internationalization ID like `multiselect.numMembers`, and then that needs to be added to `i18n/en.json` with the matching message like the other localized strings

File: components/post_view/post_attachment_opengraph/post_attachment_opengraph.jsx [hunk 2]
```diff
@@ -248,8 +248,8 @@ export default class PostAttachmentOpenGraph extends React.PureComponent {
         });
     }
 
-    isRemovePreview(post) {
-        if (post && post.props) {
+    isRemovePreview(post, currentUser) {
```
Comment (hmhealey) [inline components/post_view/post_attachment_opengraph/post_attachment_opengraph.jsx]: This doesn't look related to this change. It should be reverted if that's the case.

File: package-lock.json [hunk 3]
```diff
@@ -20,10 +20,10 @@
       "dev": true,
       "requires": {
         "@babel/types": "7.0.0-beta.44",
-        "jsesc": "^2.5.1",
-        "lodash": "^4.2.0",
-        "source-map": "^0.5.0",
-        "trim-right": "^1.0.1"
+        "jsesc": "2.5.1",
```
Comment (hmhealey) [inline package-lock.json]: What version of npm are you using? We shouldn't be changing this file too much, and I think there might be just differences between different versions of npm

Comment (lisakycho) [reply package-lock.json]: This is what I get when I run `npm version`
{ 'mattermost-webapp': '0.0.1',
  npm: '5.6.0',
  ares: '1.13.0',
  cldr: '32.0.1',
  http_parser: '2.7.0',
  icu: '60.2',
  modules: '59',
  nghttp2: '1.29.0',
  node: '9.4.0',
  openssl: '1.0.2n',
  tz: '2017c',
  unicode: '10.0',
  uv: '1.18.0',
  v8: '6.2.414.46-node.17',
  zlib: '1.2.11' }

Comment (hmhealey) [reply package-lock.json]: Since the package-lock.json changes weren't needed for this PR, you're fine for now, but it may make your life easier if you update NPM so that package-lock.json isn't modified every time `npm install` is ran

**Comment by lisakycho** [comment]
Rebased!

**Comment by cpanato** [comment]
@lisakycho can you please rebase again?

**Comment by jasonblais** [comment]
Are we waiting to merge https://github.com/mattermost/mattermost-redux/pull/508 first?

**Comment by hmhealey** [comment]
@jasonblais Yeah, that PR needs to be merged first. It just needs a rebase now since it was previously waiting on the server PR which has also been merged

**Comment by jasonblais** [comment]
@hmhealey Looks like the Redux PR is merged now

**Comment by hmhealey** [comment]
The mattermost-redux changes were merged, but this PR wasn't updated with them. If this branch is rebased onto master, it'll get the necessary redux changes.

@lisakycho Are you available to rebase this PR and fix the merge conflicts? If not, one of us can probably do it

**Comment by lisakycho** [comment]
Hey @hmhealey sorry for the late reply. I won't be able to rebase this until the end of Aug if you guys don't do it first!

**Comment by hmhealey** [comment]
Hopefully that should do it. I don't have time to do a full test right now, but Jenkins should pick up any issues

**Comment by cpanato** [comment]
hello @lisakycho thanks for this PR. there are some test build failures, can you please take a look?

you should update the snapshot tests as well

**Comment by jasonblais** [comment]
@cpanato I think Lisa mentioned she won't have time, perhaps someone on our team can help take a look at the failure?

**Comment by cpanato** [comment]
I can take a look on that @jasonblais

**Comment by cpanato** [comment]
just to confirm. @jasonblais this we should merge to master, right?

**Comment by jasonblais** [comment]
@cpanato That's correct, to master.
